drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/FieldItemViewImpl.java

```diff
@@ -98,14 +106,35 @@ public void onFieldElementClick(ClickEvent clickEvent) {
     }
 
     @Override
+    public void onFieldElementSelected() {
+        fieldElement.addClassName(ConstantHolder.SELECTED);
+        fieldItemPresenter.onFieldElementClick(this);
+    }
+
     public void onFieldElementClick() {
-        fieldElement.addClassName("selected");
+        fieldElement.addClassName(ConstantHolder.SELECTED);
+        showCheck(true);
         fieldItemPresenter.onFieldElementClick(this);
     }
 
+    @Override
+    public void showCheck(boolean show) {
+        if (show) {
+            checkElement.getStyle().setDisplay(Style.Display.BLOCK);
+        } else {
+            checkElement.getStyle().setDisplay(Style.Display.NONE);
+        }
+    }
+
+    @Override
+    public boolean isCheckShown() {
+        return !Objects.equals(Style.Display.NONE.getCssName(), checkElement.getStyle().getDisplay());
+    }
+
     @Override
     public void unselect() {
-        fieldElement.removeClassName("selected");
+        fieldElement.removeClassName(ConstantHolder.SELECTED);
```

Comment:
@yesamer 
Please use the *showCheck* method - otherwise we have two paths for the same behavior

Reply:
Done!